optimizely/event/user_event_factory.py

```diff
@@ -36,12 +39,18 @@ def create_impression_event(
       - activated_experiment is None.
     """
 
-        if not activated_experiment:
+        if not activated_experiment and flag_type is not enums.DecisionSources.ROLLOUT:
             return None
 
-        experiment_key = activated_experiment.key
-        variation = project_config.get_variation_from_id(experiment_key, variation_id)
+        if activated_experiment:
+            experiment_key = activated_experiment.key
+        else:
+            experiment_key = None
 
+        if variation_id:
+            variation = project_config.get_variation_from_id(experiment_key, variation_id)
```

Comment:
I think I had mentioned previously. There will be an error message if experiment key is None. What is being done to counter that?